agr_java_core/src/main/java/org/alliancegenome/neo4j/repository/indexer/GeneIndexerRepository.java

```diff
@@ -132,23 +136,39 @@ public void setCacheResult(Map<String, Set<String>> result) {
 	}
 
 	private class GetGeneMapThread implements Runnable {
+		File cacheFile = new File(filePrefix + "_" + getClass().getSimpleName() + ".data");
+
+		ObjectFileStorage<Map<String, Gene>> geneObjectFileStorage = new ObjectFileStorage<>();
 
 		@Override
 		public void run() {
 			log.info("Fetching genes");
-			String query = " MATCH p1=(species:Species)-[:FROM_SPECIES]-(g:Gene) ";
-
-			query += " OPTIONAL MATCH pSoTerm=(g:Gene)-[:ANNOTATED_TO]-(soTerm:SOTerm)";
-			query += " OPTIONAL MATCH p5=(g:Gene)--(:GenomicLocation) ";
-			query += " RETURN p1, pSoTerm, p5";
-
-			Iterable<Gene> genes = null;
-
-			genes = query(query);
 
 			Map<String, Gene> geneMap = new HashMap<>();
-			for (Gene gene : genes) {
-				geneMap.put(gene.getPrimaryKey(), gene);
+
+			try {
+				if (!cacheFile.exists()) {
+
+					String query = " MATCH p1=(species:Species)-[:FROM_SPECIES]-(g:Gene) ";
+					query += " OPTIONAL MATCH pSoTerm=(g:Gene)-[:ANNOTATED_TO]-(soTerm:SOTerm)";
+					query += " OPTIONAL MATCH p5=(g:Gene)--(:GenomicLocation) ";
```

Comment:
Please use text block (triple quotes) for readability.

Reply:
I was not making a change to the queries here. And I agree we do need to go through the code and make it more readable using code chunks.